examples/vision/zero_dce.py

```diff
@@ -0,0 +1,461 @@
+- The form of this curve should be as simple as possible and differentiable in the process
+of gradient backpropagation.
+
+The Light-Enhancement Curve is seperately applied to three RGB channels instead of solely on the
+illumination channel. The three-channel adjustment can better preserve the inherent color and reduce
+the risk of over-saturation. The Light-Enhancement Curve can be applied iteratively to enable more
+versatile adjustment to cope with challenging low-light conditions.
+
+![](https://li-chongyi.github.io/Zero-DCE_files/framework.png)
+
+### DCE-Net
+
+The DCE-Net is a lightweight deep neural network that learns the mapping between an input
+image and its best-fitting curve parameter maps. The input to the DCE-Net is a low-light
+image while the outputs are a set of pixel-wise curve parameter maps for corresponding
+higher-order curves. It is a plain CNN of seven convolutional layers with symmetrical
+concatenation. Each layer consists of 32 convolutional kernels of size 3×3 and stride 1
+followed by the ReLU activation function. The last convolutional layer is followed by the
+Tanh activation function, which produces 24 parameter maps for 8 iterations, where each
+iteration requires three curve parameter maps for the three channels.
+
+![](https://i.imgur.com/HtIg34W.png)
+"""
+
+
+def build_dce_net():
+    input_img = keras.Input(shape=[None, None, 3])
+    conv1 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(input_img)
+    conv2 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv1)
+    conv3 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv2)
+    conv4 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(conv3)
+    int_con1 = layers.Concatenate(axis=-1)([conv4, conv3])
+    conv5 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con1)
+    int_con2 = layers.Concatenate(axis=-1)([conv5, conv2])
+    conv6 = layers.Conv2D(
+        32, (3, 3), strides=(1, 1), activation="relu", padding="same"
+    )(int_con2)
+    int_con3 = layers.Concatenate(axis=-1)([conv6, conv1])
+    x_r = layers.Conv2D(24, (3, 3), strides=(1, 1), activation="tanh", padding="same")(
+        int_con3
+    )
+    return keras.Model(inputs=input_img, outputs=x_r)
+
+
+"""
+## Loss Functions
+
+To enable zero-reference learning in DCE-Net, we would use a set of differentiable
+non-reference losses that allow us to evaluate the quality of enhanced images.
+"""
+
+"""
+### Color Constancy Loss
+
+The Color Constancy Loss is used to correct the potential color deviations in the
+enhanced image and also build the relations among the three adjusted channels.
+"""
+
+
+def color_constancy_loss(x, **kwargs):
+    mean_rgb = tf.reduce_mean(x, kwargs.get("axis", (1, 2)), keepdims=True)
```

Comment:
Let me explain what I meant.

In the statement, `tf.reduce_mean(x, (1, 2))`, the reader doesn't know what `(1, 2)` means. So the statement is not very readable / clear.

For this reason, it's better to use `tf.reduce_mean(x, axis=(1, 2))`. Then the reader knows what is going on: the axes of reduction are being specified. It's more explicit.

In the first case you're passing the argument by position. In the second case you're passing it as a keyword argument (the "keyword" being "axis").

Reply:
Thanks for the explanation. I used keyword args for better readability.